keps/sig-node/5055-device-taints-and-tolerations/README.md

```diff
@@ -0,0 +1,971 @@
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+<!--
+This section is incredibly important for producing high-quality, user-focused
+documentation such as release notes or a development roadmap. It should be
+possible to collect this information before implementation begins, in order to
+avoid requiring implementors to split their attention between writing release
+notes and implementing the feature itself. KEP editors and SIG Docs
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+
+With Dynamic Resource Allocation (DRA), DRA drivers publish information about
+the devices that they manage in ResourceSlices. This information is used by the
+scheduler when selecting devices for user requests in ResourceClaims.
+
+With this KEP, DRA drivers can mark devices as tainted such that they won't be
+used for scheduling new pods. In addition, pods already running with access to
+a tainted device can be stopped automatically. Cluster administrators can do
+the same by creating a
+[ResourceSliceOverride](../5027-dra-admin-controlled-device-attributes] with a
+taint.
+
+Users can decide to ignore specific taints by adding tolerations to their
+ResourceClaim.
+
+## Motivation
+
+
+### Goals
+
+- Enable taking devices offline for maintenance while still allowing tests pods
+  to request and use those devices.
+
+- Enable users to decide whether they want to keep running a workload in a degraded
+  mode while a device is unhealthy or prefer to get pods rescheduled.
+
+
+### Non-Goals
+
+## Proposal
+
+#### Degraded Devices
+
+As driver itself can detect problems which may or not be tolerable for
+workloads, like degraded performance due to overheating. Removing such devices
+from the ResourceSlice would unconditionally prevent using them for new
+pods. Instead, publishing with a taint informs users about this degradation and
+leaves them the choice whether the device is still usable enough to run pods.
+It also automates stopping pods which don't tolerate such a degradation.
+
+#### External Health Monitoring
+
+As cluster admin, I am deploying a vendor-provided DRA driver together with a
+separate monitoring component for hardware aspects that are not available or
+not supported by that DRA driver. When that component detects problems, it can
+check its policy configuration and decide to take devices offline by creating
+a ResourceSliceOverride with a taint for affected devices.
```

Comment:
Probably, we could add maintenance use-case??

Reply:
This is covered by the first goal. I'm not sure whether it's worth having a separate user story about it.

Reply:
Oh, sorry. I found it 🙇